server/schedulers/hot_region.go

```diff
@@ -1159,33 +1165,41 @@ func (h *hotScheduler) GetPendingInfluence() map[uint64]*Influence {
 	return ret
 }
 
-// calcPendingWeight return the calculate weight of one Operator, the value will between [0,1]
-func (h *hotScheduler) calcPendingWeight(op *operator.Operator) float64 {
-	if op.CheckExpired() || op.CheckTimeout() {
-		return 0
-	}
+// calcPendingInfluence return the calculate weight of one Operator, the value will between [0,1]
+func (h *hotScheduler) calcPendingInfluence(op *operator.Operator, maxZombieDur time.Duration) (weight float64, needGC bool) {
 	status := op.Status()
+	switch {
+	case op.CheckExpired():
+		status = operator.EXPIRED
```

Comment:
CheckExpired/CheckTimeout will update the status.

Reply:
Is there any problem, shouldn't it be updated?

Reply:
I mean it will be updated inside the CheckExpired/CheckTimeout.